src/puppetlabs/kitchensink/core.clj

```diff
@@ -141,6 +141,14 @@ to be a zipper."
       (> num-strings 3)
       (str (first strings) ", " (to-sentence (rest strings))))))
 
+(defn key->str
+  "Convert a keyword to a string, stripping the leading : character. This is
+  distinct from the `name` function as it will preserve the 'namespace' portion
+  before a slash. If the key is already a string, it is returned unmodified."
+  [kw]
+  (if (keyword? kw)
+    (subs (str kw) 1)
+    kw))
```

Comment:
Since this only checks whether the input is a keyword, would it make sense to return `(name kw)` in the case where it's not a keyword? @donoghuc mentioned in https://github.com/puppetlabs/clj-kitchensink/pull/111 that `name` behavior should generally be preserved, like handling `nil` or numeric values.

Reply:
I don't think that would make the function more correct and, if anything, would be apt to lead to bugs. The only other supported type for `name` is Symbol, and it has the same surprising behavior for symbols as it does for keywords (removing the namespace). I don't think there's a real use case for symbols, but if there ever were I would prefer it just be that this is not the function to use rather than have it _appear_ to work but actually have incorrect behavior.